tests/integration/clientv3/examples/example_watch_test.go

```diff
@@ -0,0 +1,157 @@
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   exampleEndpoints(),
+			DialTimeout: dialTimeout,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer cli.Close()
+
+		rch := cli.Watch(context.Background(), "foo")
+		for wresp := range rch {
+			for _, ev := range wresp.Events {
+				fmt.Printf("%s %q : %q\n", ev.Type, ev.Kv.Key, ev.Kv.Value)
+			}
+		}
+	})
+	// PUT "foo" : "bar"
+}
+
+func mockWatcher_watchWithPrefix() {
+	fmt.Println(`PUT "foo1" : "bar"`)
+}
+
+func ExampleWatcher_watchWithPrefix() {
+	forUnitTestsRunInMockedContext(mockWatcher_watchWithPrefix, func() {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   exampleEndpoints(),
+			DialTimeout: dialTimeout,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer cli.Close()
+
+		rch := cli.Watch(context.Background(), "foo", clientv3.WithPrefix())
+		for wresp := range rch {
+			for _, ev := range wresp.Events {
+				fmt.Printf("%s %q : %q\n", ev.Type, ev.Kv.Key, ev.Kv.Value)
+			}
+		}
+	})
+	// PUT "foo1" : "bar"
+}
+
+func mockWatcher_watchWithRange() {
+	fmt.Println(`PUT "foo1" : "bar1"`)
+	fmt.Println(`PUT "foo2" : "bar2"`)
+	fmt.Println(`PUT "foo3" : "bar3"`)
+}
+
+func ExampleWatcher_watchWithRange() {
+	forUnitTestsRunInMockedContext(mockWatcher_watchWithRange, func() {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   exampleEndpoints(),
+			DialTimeout: dialTimeout,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+		defer cli.Close()
+
+		// watches within ['foo1', 'foo4'), in lexicographical order
+		rch := cli.Watch(context.Background(), "foo1", clientv3.WithRange("foo4"))
+
+		go func() {
+			cli.Put(context.Background(), "foo1", "bar1")
+			cli.Put(context.Background(), "foo5", "bar5")
+			cli.Put(context.Background(), "foo2", "bar2")
+			cli.Put(context.Background(), "foo3", "bar3")
+		}()
+
+		i := 0
+		for wresp := range rch {
+			for _, ev := range wresp.Events {
+				fmt.Printf("%s %q : %q\n", ev.Type, ev.Kv.Key, ev.Kv.Value)
+				i++
+				if i == 3 {
+					// After 3 messages we are done.
+					cli.Delete(context.Background(), "foo", clientv3.WithPrefix())
+					cli.Close()
+					return
+				}
+			}
+		}
+	})
+
+	// Output:
+	// PUT "foo1" : "bar1"
+	// PUT "foo2" : "bar2"
+	// PUT "foo3" : "bar3"
+}
+
+func mockWatcher_watchWithProgressNotify() {
+	fmt.Println(`wresp.Header.Revision: 0`)
+	fmt.Println(`wresp.IsProgressNotify: false`)
+}
+
+func ExampleWatcher_watchWithProgressNotify() {
+	forUnitTestsRunInMockedContext(mockWatcher_watchWithProgressNotify, func() {
+		cli, err := clientv3.New(clientv3.Config{
+			Endpoints:   exampleEndpoints(),
+			DialTimeout: dialTimeout,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		rch := cli.Watch(context.Background(), "foo", clientv3.WithProgressNotify())
+		go func() {
+			time.Sleep(time.Second)
+			err := cli.Close()
+			if err != nil {
+				log.Fatal(err)
+			}
+		}()
+		wresp := <-rch
+		fmt.Printf("wresp.Header.Revision: %d\n", wresp.Header.Revision)
```

Comment:
I think we can get rid of the comparison of `Revision` here. Since we have a shared cluster for all the example tests. This comparison becomes very flaky depending on which examples are run before this example.

Reply:
Thank you for the suggestion. Indeed that fixed the test.